lib/collections/schemas/orders.js

```diff
@@ -156,34 +161,46 @@ registerSchema("OrderTransaction", OrderTransaction);
  * @property {OrderTransaction[]} transactions optional
  */
 export const Order = new SimpleSchema({
-  userId: {
+  "userId": {
     type: String,
     unique: false
   },
-  cartId: {
+  "cartId": {
     type: String,
     optional: true
   },
-  history: {
-    type: [History],
+  "history": {
+    type: Array,
     optional: true
   },
-  documents: {
-    type: [Document],
+  "history.$": {
+    type: History
+  },
+  "documents": {
+    type: Array,
     optional: true
   },
-  notes: {
-    type: [Notes],
+  "documents.$": {
+    type: Document
+  },
+  "notes": {
+    type: Array,
     optional: true
   },
-  items: {
-    type: [OrderItem],
+  "notes.$": Notes,
+  "items": {
+    type: Array,
     optional: true
   },
-  transactions: {
-    type: [OrderTransaction],
+  "items.$": CartItem.clone().extend(OrderItem),
+  "transactions": {
+    type: Array,
     optional: true
-  }
-});
+  },
+  "transactions.$": OrderTransaction
+}, { check, tracker: Tracker });
 
 registerSchema("Order", Order);
+
+export const OrderDocument = Cart.clone().extend(Order).extend(OrderItem);
+registerSchema("OrderDocument", OrderDocument);
```

Comment:
This was formerly handled by passing an array of schemas to `attachSchema` but that no longer works. So now I'm registering it as its own schema and then attaching that.